third_party/amd/lib/TritonAMDGPUToLLVM/TargetInfo.cpp

```diff
@@ -126,10 +126,146 @@ Value TargetInfo::programId(RewriterBase &rewriter, Location loc,
   return LLVM::AMD::llGetPid(loc, rewriter, moduleOp, axis);
 }
 
+static std::optional<Operation *> getReduxOpForDPP(triton::ReduceOp op) {
+  // In order to utilize the full power of DPP modifier, the redux operator
+  // should be a single arthmetic operator.
+  //
+  // Reference: third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/TargetInfo.cpp
+  if (op.getNumOperands() != 1 || op.getNumResults() != 1)
+    return std::nullopt;
+  Block *block = &(*op.getCombineOp().begin());
+  Operation *yield = block->getTerminator();
+  Operation *reduceOp = yield->getOperand(0).getDefiningOp();
+  if (!reduceOp || reduceOp->getNumOperands() != 2 ||
+      reduceOp->getNumResults() != 1)
+    return std::nullopt;
+  if (reduceOp->getOperand(0) != block->getArgument(0) ||
+      reduceOp->getOperand(1) != block->getArgument(1))
+    return std::nullopt;
+
+  if (isa<arith::AddIOp>(reduceOp) || isa<arith::AddFOp>(reduceOp) ||
```

Comment:
I think `isa` supports multiple cases in `<...>`. Also could you put comment to explain why this list? I guess it's somewhere in the ISA doc; you can give a section number here.

Reply:
At the beginning this was meant to filter out the ops that don't have their corresponding dpp instructions. But since it can still have some improvement without fusing into a dpp instruction, it should be fine to remove this part.